libs/vm-modules/src/math/random.cpp

```diff
@@ -96,8 +96,8 @@ IfIsPtrFixed128<T, Ptr<T>> Rand(VM *vm, Ptr<T> const &a, Ptr<T> const &b)
 
   auto a_dbl = static_cast<double>(a->data_);
   auto b_dbl = static_cast<double>(b->data_);
-  auto x =
-      static_cast<fixed_point::fp128_t>(std::uniform_real_distribution<double>{a_dbl, b_dbl}(mt));
+  auto x     = fetch::math::AsType<fixed_point::fp128_t>(
+      std::uniform_real_distribution<double>{a_dbl, b_dbl}(mt));
```

Comment:
this method is not safe anyway right? (i.e. AsFP128 will return a random FP128 constructed via double). Do we have a Jira task for fixing this? If not can you please raise one.

Reply:
we would have to replace the whole RNG used in this case instead of the std one. I'm fine with that and it would also be more deterministic -eg. in case C++ spec decide to use another RNG underneath. I will create a ticket for it.

Reply:
Ml-436 added.